### PR TITLE
[Shepherd] Do stricter checking of -D command line arguments

### PR DESCRIPTION
**Pull request by CodaFi**
This rebases #6813 and updates a failing test.

Fixes [SR-2404](https://bugs.swift.org/browse/SR-2404).

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci please smoke test

File: test/Driver/options-repl.swift [hunk 1]
```diff
@@ -16,15 +16,15 @@
 
 
 // RUN: %swift_driver -lldb-repl -### | %FileCheck -check-prefix=LLDB %s
-// RUN: %swift_driver -lldb-repl -DA,B,C -DD -L /path/to/libraries -L /path/to/more/libraries -F /path/to/frameworks -lsomelib -framework SomeFramework -sdk / -I "this folder" -module-name Test -target %target-triple -### | %FileCheck -check-prefix=LLDB-OPTS %s
+// RUN: %swift_driver -lldb-repl -DD -L /path/to/libraries -L /path/to/more/libraries -F /path/to/frameworks -lsomelib -framework SomeFramework -sdk / -I "this folder" -module-name Test -target %target-triple -### | %FileCheck -check-prefix=LLDB-OPTS %s
```
Comment (jrose-apple) [inline test/Driver/options-repl.swift]: Please do still include multiple -D flags. It's not critical but the test was confirming that they get passed in the same order.

**Review by jrose-apple — COMMENTED** [review]
Looks good with just one testing comment.

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**